src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -461,44 +414,153 @@ seastar::future<nlohmann::json> HTTPProxy::_handleQuery(nlohmann::json&& jsonReq
+        .name = request.metadata.name,
+        .hashScheme = static_cast<dto::HashScheme>(request.metadata.hashScheme),
+        .storageDriver = static_cast<dto::StorageDriver>(request.metadata.storageDriver),
+        .capacity={
+            .dataCapacityMegaBytes = request.metadata.capacity.dataCapacityMegaBytes,
+            .readIOPs = request.metadata.capacity.readIOPs,
+            .writeIOPs = request.metadata.capacity.writeIOPs,
+            .minNodes = request.metadata.capacity.minNodes,
+        },
+        .retentionPeriod = request.metadata.retentionPeriod,
+        .heartbeatDeadline = request.metadata.heartbeatDeadline,
+        .deleted = request.metadata.deleted
+    };
+
+    std::vector<String> rends;
+    for(auto& re: request.rangeEnds) {
+        rends.push_back(re);
+    }
+
+    return _client.makeCollection(std::move(meta), std::move(rends))
+        .then([](Status&& status) {
+            return MakeHTTPResponse<sh::dto::CollectionCreateResponse>(sh::Status{.code=status.code, .message=status.message}, sh::dto::CollectionCreateResponse{});
+        });
+}
+
+seastar::future<std::tuple<sh::Status, sh::dto::CreateSchemaResponse>>
+HTTPProxy::_handleCreateSchema(sh::dto::CreateSchemaRequest&& request) {
+    K2LOG_D(log::httpproxy, "Received create schema request {}", request);
+
+    std::vector<k2::dto::SchemaField> k2fields;
+
+    for (auto& f: request.schema.fields) {
+        k2fields.push_back(k2::dto::SchemaField{
+            .type = static_cast<k2::dto::FieldType>(f.type),
+            .name = f.name,
+            .descending = f.descending,
+            .nullLast = f.nullLast
+        });
+    }
+    k2::dto::Schema schema{
+        .name = request.schema.name,
+        .version = request.schema.version,
+        .fields = std::move(k2fields),
+        .partitionKeyFields = request.schema.partitionKeyFields,
+        .rangeKeyFields = request.schema.rangeKeyFields
+    };
+
+    return _client.createSchema(String(request.collectionName), std::move(schema))
+        .then([](auto&& result) {
+            return MakeHTTPResponse<sh::dto::CreateSchemaResponse>(sh::Status{.code=result.status.code, .message=result.status.message}, sh::dto::CreateSchemaResponse{});
+        });
+}
+
+seastar::future<std::tuple<sh::Status, sh::dto::BeginTxnResponse>>
+HTTPProxy::_handleBeginTxn(sh::dto::BeginTxnRequest&& request){
+    K2LOG_D(log::httpproxy, "Received begin txn request {}", request);
+    k2::K2TxnOptions opts{
+        .deadline= k2::Deadline<>(request.options.timeout),
+        .priority = static_cast<k2::TxnPriority>(request.options.priority),
+        .syncFinalize = request.options.syncFinalize
+    };
+    return _client.beginTxn(std::move(opts))
+        .then([this](auto&& txn) {
+            K2LOG_D(k2::log::httpproxy, "begin txn: {}", txn.mtr());
+            auto ts = txn.mtr().timestamp;
+            sh::dto::Timestamp shts(ts.tEndTSECount(), ts.tEndTSECount() - ts.tStartTSECount(), ts.tsoId());
```

Comment:
In   Timestamp ctor https://github.com/futurewei-cloud/chogori-platform/blob/master/src/k2/dto/Timestamp.h#L47, tsoId is the second parameter
    `Timestamp(uint64_t tEndTSECount, uint32_t tsoId, uint32_t tStartDelta);`
But here it's the third.

Reply:
good catch! I'll change the Timestamp so it doesn't allow this sort of mistake

Reply:
done